mobile-ui/src/main/java/com/playone/mobile/ui/Navigator.kt

```diff
@@ -15,9 +19,30 @@ class Navigator constructor(activityContext: Context) {
      * navigate to a Fragment with [AppCompatActivity] and the transactions
      * in [FragmentTransaction]
      */
-    fun navigateTo(context: AppCompatActivity, transactions: FragmentTransaction.() -> Unit) {
+    fun navigateToFragment(
+        context: AppCompatActivity,
+        transactions: FragmentTransaction.() -> Unit
+    ) {
         context.transact {
             transactions()
         }
     }
 }
+
+inline fun <reified T : AppCompatActivity> Navigator.navigateToActivity(
+    context: AppCompatActivity,
+    crossinline intent: Intent.() -> Unit = {}
+) = context.start<T> {
+    intent()
+}
+
+inline fun <reified T : AppCompatActivity> Navigator.navigateToActivity(
+    fragment: Fragment,
+    crossinline intent: Intent.() -> Unit = {}
+) = (fragment.activity)?.let {
+    (it is AppCompatActivity).ifTrue {
```

Comment:
```kotlin
(it as? AppCompatActivity).takeif { it.start<T> { intent() } }
```

will be better.

Reply:
`takeif` require `()->Boolean`

Reply:
I didn't check the code 😞  I focused on casting, there're too many nests, it could be easier and more elegant. 👍 

```kotlin
 (fragment.activity as? AppCompatActivity).let { it?.start<T>(intent) }
```